onmt/dynamic/transforms/mics.py

```diff
@@ -0,0 +1,97 @@
+from onmt.utils.logging import logger
+from onmt.dynamic.transforms import register_transform
+from .transform import Transform
+
+
+@register_transform(name='filtertoolong')
+class FilterTooLongTransform(Transform):
+    """Filter out sentence that are too long."""
+
+    def __init__(self, opts):
+        super().__init__(opts)
+        self.src_seq_length = opts.src_seq_length
+        self.tgt_seq_length = opts.tgt_seq_length
+
+    @classmethod
+    def add_options(cls, parser):
+        """Avalilable options relate to this Transform."""
+        group = parser.add_argument_group('Transform/Filter')
+        group.add('--src_seq_length', '-src_seq_length', type=int, default=200,
+                  help="Maximum source sequence length")
+        group.add('--tgt_seq_length', '-tgt_seq_length', type=int, default=200,
+                  help="Maximum target sequence length")
+
+    def apply(self, example, is_train=False, stats=None, **kwargs):
+        """Return None if too long else return as is."""
+        if (len(example['src']) > self.src_seq_length or
+                len(example['tgt']) > self.tgt_seq_length):
+            if stats is not None:
+                stats.filter_too_long()
+            return None
+        else:
+            return example
+
+    def _repr_args(self):
+        """Return str represent key arguments for class."""
+        return '{}={}, {}={}'.format(
+            'src_seq_length', self.src_seq_length,
+            'tgt_seq_length', self.tgt_seq_length
+        )
+
+
+@register_transform(name='prefix')
+class PrefixSrcTransform(Transform):
```

Comment:
This requires adding in every corpus the definition the src_prefix/tgt_prefix you've mentioned, a little bit annoying I think.

Reply:
That's the idea yes. Not more annoying than adding src_lang, tgt_lang (that could be made optional by the way).
And this would be a bit more flexible to simply allow user to set the prefix tags they need instead of having a set format.
Or, this whole 'custom prefix' could be another transform. Not sure.

Reply:
The required prefix is addressed in #18.

Reply:
Thanks! I think it's a bit more flexible that way :)
And if we prefer to have an "automatic" prefix transform like before, we can always add it later or use it as a custom one.